### DISCUSSION TITLE
Version Selector

### DISCUSSION THREAD

**Discussion by Diegiwg** [pacstall/pacstall-programs#3787] [Ideas]
As far as I know, **pacscripts** only support one version, which is usually the latest release.

The idea would be (if the maintainer wishes), to build a list of versions in a pacscript.

We can have the following syntax: `pacstall -I package -v 1.0.0`. It would install version 1.0.0 of the package, where: `pacstall -I package` would install version 5.4.0, which is the latest version of the package.

Inside the **pacscript**, we can have the following syntax:

``` shell
pkgver() {
case "${USER_VERSION}" in
"1.0.0.")
url="https://example.com/package/linux_${USER_VERSION}_amd64.deb"
hash="48e9a77015695fe69390ac99b54184cf793f31c3452d174fcf41052ccb2ae211"
;;
"3.0.0.")
url="https://example.com/package/linux_${USER_VERSION}_amd64.deb"
hash="48e9a77015695fe69390ac99b54184cf793f31c3452d174fcf41052ccb2ae211"
;;
"6.0.0.")
url="https://example.com/package/v${USER_VERSION}/linux_${USER_VERSION}_amd64.deb"
hash="48e9a77015695fe69390ac99b54184cf793f31c3452d174fcf41052ccb2ae211"
;;
*) return 1 ;;
esac
}
version="$(pkgver)"
```

Would it be possible to do this?

**Comment by Diegiwg** [comment]
ping : @Henryws

**Comment by Elsie19** [comment]
If you want, you can check `/var/cache/pacstall/${name}/${version}/${name}.pacscript` if you want older versions. Is that what you were after?

**Comment by Diegiwg** [comment]
No, the idea was to be a feature that lets you choose between available versions, like:

`pacstall -I package -v 1.0.0`
`pacstall -I package -v 2.0.0`
`pacstall -I package -v 3.0.0`

Assuming `package.pacscript` is exposing as available versions the 'array' `('1.0.0' '2.0.0' '3.0.0')`.

Oh like, if the person did: `pacstall --query-info package`. The available versions are returned.

**Reply by Elsie19** [reply]
What would the use-case of this be?

**Comment by Diegiwg** [comment]
I realized that if an old version is useful (like Java and the like), we can just have a "$name-$version-deb" package. So my idea is not valid. Thank you for your attention.
